packages/block-editor/src/components/iframe/index.js

```diff
@@ -28,17 +27,17 @@ const BODY_CLASS_NAME = 'editor-styles-wrapper';
 const BLOCK_PREFIX = 'wp-block';
 
 /**
- * Clones stylesheets targetting the editor canvas to the given document. A
- * stylesheet is considered targetting the editor a canvas if it contains the
- * `editor-styles-wrapper`, `wp-block`, or `wp-block-*` class selectors.
+ * Hook that clones stylesheets targetting the editor canvas to the given
+ * document. A stylesheet is considered targetting the editor a canvas if it
+ * contains the `editor-styles-wrapper`, `wp-block`, or `wp-block-*` class
+ * selectors.
  *
- * Ideally, this hook should be removed in the future and styles should be added
- * explicitly as editor styles.
- *
- * @param {Document} doc The document to append cloned stylesheets to.
+ * This is a compatibility layer. Ideally, this hook should be removed in the
+ * future and styles should be added explicitly as editor styles.
  */
-function styleSheetsCompat( doc ) {
+function useEditorStylesCompat() {
 	// Search the document for stylesheets targetting the editor canvas.
+	const clonedStyles = [];
 	Array.from( document.styleSheets ).forEach( ( styleSheet ) => {
```

Comment:
Can we move all this into `useRefEffect`? It would also make it much easier to see the essence of this PR.

Reply:
We can, but I believe it'd be less performant. The iteration through the root document styles doesn't have to be executed on changes to the iframe node.